src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const finalObject = [];
```

Comment:
The name `finalObject` is a bit misleading as it suggests a single object, whereas it is actually an array of state objects. A more descriptive name like `stateHistory` or `states` would be better.